lib/service/proxy_settings.go

```diff
@@ -76,7 +76,7 @@ func (p *proxySettings) buildProxySettings(proxyListenerMode types.ProxyListener
 		proxySettings.DB.MySQLListenAddr = p.cfg.Proxy.MySQLAddr.String()
 	}
 	if p.cfg.Proxy.Kube.Enabled {
-		proxySettings.Kube.ListenAddr = p.getProxyKubeAddress(proxyListenerMode)
+		proxySettings.Kube.ListenAddr = p.cfg.Proxy.Kube.ListenAddr.String()
```

Comment:
restore the https://github.com/gravitational/teleport/blob/branch/v7/lib/service/service.go#L2614 behaviour to support connection from tsh v7 to teleport v8 running in multiplex mode.

Reply:
@smallinsky Hm, does this work for v8 tsh clients too? What if kube listen addr is not specified? Or will the v8 client select the web proxy endpoint?

Reply:
Yes, the v8 `tsh` client will select the WebPort if TLS Routing is configured supported by the Proxy.